Rolique/Model/Helpers/ImageManager.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  ImageManager.swift
+//  Rolique
+//
+//  Created by Maksym Ivanyk on 9/2/19.
+//  Copyright © 2019 Rolique. All rights reserved.
+//
+
+import UIKit
+import Foundation
+
+final class ImageManager {
+  private lazy var observer: NSObjectProtocol = {
+    return NotificationCenter.default.addObserver(forName: UIApplication.didReceiveMemoryWarningNotification, object: nil, queue: nil) { [weak self] notification in
+      self?.clearImagesFolder()
+    }
+  }()
+  
+  static let shared = ImageManager()
+  
+  private init() {}
+  
+  deinit {
+    NotificationCenter.default.removeObserver(self.observer)
+  }
+  
+  private let fileManager = FileManager.default
+  private func documentDirectoryURL() throws -> URL {
+    return try fileManager.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: false)
+  }
+  private func imagesDirectoryURL() throws -> URL {
+    return URL(fileURLWithPath: try documentDirectoryURL().path + "/Images")
+  }
+  
+  func getImage(with url: URL) -> UIImage? {
+    do {
+      let imagesDirectory = try imagesDirectoryURL()
+      if let data = fileManager.contents(atPath: imagePath(imagesDirectoryURL: imagesDirectory, imageURL: url)) {
+        return UIImage(data: data, scale: UIScreen.main.scale)
+      }
+    } catch let error as NSError {
+      print("Could not get image: \(error.debugDescription), for url: \(url)")
+    }
+    return nil
+  }
+  
+  func saveImage(data: Data, forURL url: URL) {
+    do {
+      let imagesDirectory = try imagesDirectoryURL()
+      checkDirectoryExists(with: imagesDirectory)
+      fileManager.createFile(atPath: imagePath(imagesDirectoryURL: imagesDirectory, imageURL: url),
+                             contents: data,
+                             attributes: nil)
+    } catch let error as NSError {
+      print("Could not save image: \(error.debugDescription), for url: \(url)")
```

Comment:
It's a bad practise to leave entire catch block empty. If you don't plan to handle error just write try?